runners/flink/src/main/java/org/apache/beam/runners/flink/adapter/BeamAdapterUtils.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.adapter;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.runners.flink.translation.types.CoderTypeInformation;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.CannotProvideCoderException;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.IterableCoder;
+import org.apache.beam.sdk.coders.MapCoder;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.util.construction.CoderTranslation;
+import org.apache.beam.sdk.util.construction.RehydratedComponents;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+
+public class BeamAdapterUtils {
+  private BeamAdapterUtils() {}
+
+  @SuppressWarnings("nullness")
```

Comment:
Local checks/assertions are better, because they don't mask errors that might be introduced as the code is edited. We should aim for no broken windows / zero suppressions. Certainly in new code. Any suppression should be linked to a bug with an explanation for why it cannot be addressed.

Using `org.apache.beam.sdk.util.Preconditions.checkArgumentNotNull` or `org.apache.beam.sdkutil.Preconditions.checkStateNotNull` are both good options. They convert NPE at some random point in code to an IllegalArgumentException or IllegalStateException at the point of detection.